src/posix/platform/uart.cpp

```diff
@@ -237,8 +237,8 @@ static void InitializeSessionSocket(void)
     // if we have SO_NOSIGPIPE, then set it. Otherwise, we're going
     // to simply ignore it.
 #if defined(SO_NOSIGPIPE)
-    rval = 1;
-    VerifyOrExit((rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval)) != -1, OT_NOOP));
+    rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval));
+    VerifyOrExit(rval != -1, OT_NOOP);
```

Comment:
> Should we close(sSessionSocket) before sSessionSocket = -1 after exit: ?

I think yes and it needs to be refactored. Will do in a separate PR.

> Should we ignore this setsockopt(SO_NOSIGPIPE) (i.e. not set rval) even if it failed because it seems non-critical according to the comments above?

I think if `SO_NOSIGPIPE ` is defined, we expect `setsockopt(... SO_NOSIGPIPE ...)` to success. The comment means that we can just not set it if `SO_NOSIGPIPE ` is not defined.

Reply:
See https://github.com/openthread/openthread/pull/5405.